src/tacacs/nss/patch/0012-Write-error-message-to-syslog-when-add-user-failed.patch

```diff
@@ -0,0 +1,40 @@
+From 3f88c7fe559a004b580fd515f27a1cb3e06a32f9 Mon Sep 17 00:00:00 2001
+From: liuh-80 <liuh@microsoft.com>
+Date: Wed, 23 Aug 2023 11:18:45 +0800
+Subject: [PATCH] Write error message to syslog when add user failed or connect to TACACS server failed.
+
+---
+ nss_tacplus.c | 10 ++++------
+ 1 file changed, 4 insertions(+), 6 deletions(-)
+
+diff --git a/nss_tacplus.c b/nss_tacplus.c
+index c22d6bd..f2ba035 100644
+--- a/nss_tacplus.c
++++ b/nss_tacplus.c
+@@ -540,8 +540,7 @@ static int create_or_modify_local_user(const char *name, int level, bool existin
+             snprintf(sgid, 10, "%d", user->gid);
+             snprintf(home, 63, "/home/%s", name);
+             if(0 != user_mod_add(command, name, sgid, user->secondary_grp, user->info, home, user->shell)) {
+-                if(debug)
+-                    syslog(LOG_ERR, "%s: %s %s failed", nssname, command, name);
++                 syslog(LOG_ERR, "%s: %s %s failed", nssname, command, name);
+                  delete_conf_line(name);
+                  return -1;
+              }
+@@ -858,10 +857,9 @@ lookup_tacacs_user(struct pwbuf *pb)
+         arep.status = TAC_PLUS_AUTHOR_STATUS_ERROR; /* if author_send fails */
+         tac_fd = connect_tacacs(&attr, srvr);
+         if (tac_fd < 0) {
+-            if(debug)
+-                syslog(LOG_WARNING, "%s: failed to connect TACACS+ server %s,"
+-                    " ret=%d: %m", nssname, tac_srv[srvr].addr ?
+-                    tac_ntop(tac_srv[srvr].addr->ai_addr) : "unknown", tac_fd);
++            syslog(LOG_ERR, "%s: failed to connect TACACS+ server %s,"
```

Comment:
Better to change existing patch file `src/tacacs/nss/patch/0001-Modify-user-map-profile.patch`

Reply:
Fixed.